src/matcha_ml/state/matcha_state.py

```diff
@@ -114,13 +125,27 @@ class MatchaStateService:
 
     matcha_state_path = MATCHA_STATE_PATH
 
-    def __init__(self) -> None:
+    def __init__(
```

Comment:
Nit: throw an error for the edge-case that both optional args are provided and are not consistent with each other? If checking they're consistent is a headache then could just mention in the docstring which takes priority.

Reply:
Really good point

Reply:
Updated